streampipes-extensions/streampipes-processors-geo-jvm/src/main/java/org/apache/streampipes/processors/geo/jvm/jts/processor/derivedgeometry/polygon/CreateDerivedPolygonProcessor.java

```diff
@@ -0,0 +1,139 @@
+  private static final Logger LOG = LoggerFactory.getLogger(CreateDerivedPolygonProcessor.class);
+
+  @Override
+  public DataProcessorDescription declareModel() {
+    return ProcessingElementBuilder.create(
+            "org.apache.streampipes.processors.geo.jvm.jts.processor.derivedgeometry.polygon")
+        .category(DataProcessorType.GEO)
+        .withAssets(Assets.DOCUMENTATION, Assets.ICON, "derivedPolygon.png")
+        .withLocales(Locales.EN)
+        .requiredStream(StreamRequirementsBuilder
+            .create()
+            .requiredPropertyWithUnaryMapping(
+                EpRequirements.domainPropertyReq("http://www.opengis.net/ont/geosparql#Geometry"),
+                Labels.withId(GEOM_KEY),
+                PropertyScope.MEASUREMENT_PROPERTY)
+            .requiredPropertyWithUnaryMapping(
+                EpRequirements.domainPropertyReq("http://data.ign.fr/def/ignf#CartesianCS"),
+                Labels.withId(EPSG_KEY),
+                PropertyScope.MEASUREMENT_PROPERTY)
+            .build())
+        .requiredSingleValueSelection(Labels.withId(POLYGON_OUTPUT_TYPE_KEY),
+            Options.from("Bounding Box", "Convex Hull"))
+        .outputStrategy(OutputStrategies.append(
+                EpProperties.stringEp(
+                    Labels.withId(DERIVED_GEOM_KEY),
+                    DERIVED_GEOM_RUNTIME,
+                    "http://www.opengis.net/ont/geosparql#Geometry"),
+                EpProperties.integerEp(
+                    Labels.withId(DERIVED_EPSG_KEY),
+                    DERIVED_EPSG_RUNTIME,
+                    "http://data.ign.fr/def/ignf#CartesianCS")
+            )
+        )
+        .build();
+  }
+
+  @Override
+  public void onInvocation(ProcessorParams parameters, SpOutputCollector spOutputCollector,
+                           EventProcessorRuntimeContext runtimeContext) throws SpRuntimeException {
+
+    this.geometryMapper = parameters.extractor().mappingPropertyValue(GEOM_KEY);
+    this.epsgMapper = parameters.extractor().mappingPropertyValue(EPSG_KEY);
+    this.outputType = parameters.extractor().selectedSingleValue(POLYGON_OUTPUT_TYPE_KEY, String.class);
+    this.params = parameters;
+  }
+
+  @Override
+  public void onEvent(Event event, SpOutputCollector collector) throws SpRuntimeException {
+    String geom = event.getFieldBySelector(geometryMapper).getAsPrimitive().getAsString();
+    Integer epsg = event.getFieldBySelector(epsgMapper).getAsPrimitive().getAsInt();
+    Geometry geometry = SpGeometryBuilder.createSPGeom(geom, epsg);
+    if (geometry instanceof Point) {
+      SpMonitoringManager.INSTANCE.addErrorMessage(params.getGraph().getElementId(),
+          SpLogEntry.from(System.currentTimeMillis(),
+              StreamPipesErrorMessage.from(new SpNotSupportedGeometryException(
+                  "Point Geometry is not supported"))));
+    } else {
+      Polygon derivedPolygonOutput = null;
+      switch (this.outputType) {
+        case "Bounding Box":
```

Comment:
Since you use these terms multiple times you can think of using a variable or defining an enum